pom.xml

```diff
@@ -42,26 +42,34 @@ THE SOFTWARE.
       <comments>All source code is under the MIT license.</comments>
     </license>
   </licenses>
-
+    
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <compileSource>1.7</compileSource>
+    <compileTarget>1.7</compileTarget>
```

Comment:
I think your changes can be compiled with 1.6, and you don't need to target 1.7.
1.596.1 still works with Java6.